client/src/main/java/ch/bifrost/client/impl/keyexchange/DatagramVerifyingConverter.java

```diff
@@ -0,0 +1,134 @@
+package ch.bifrost.client.impl.keyexchange;
+
+import java.io.IOException;
+import java.io.ByteArrayInputStream;
+import java.util.concurrent.TimeUnit;
+
+import com.google.common.base.Optional;
+
+import java.security.cert.CertPath;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.PKIXParameters;
+import java.security.cert.CertPathParameters;
+import java.security.cert.CertPathValidator;
+import java.security.cert.CertPathValidatorException;
+import java.security.cert.PKIXCertPathValidatorResult;
+
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.Signature;
+import java.security.SignatureException;
+
+import org.bouncycastle.asn1.ASN1Sequence;
+import org.bouncycastle.asn1.ASN1BitString;
+
+import ch.bifrost.core.api.datagram.CounterpartAddress;
+import ch.bifrost.core.api.datagram.DatagramEndpoint;
+import ch.bifrost.core.api.datagram.DatagramMessage;
+import ch.bifrost.core.impl.datagram.InvalidDatagramException;
+
+public class DatagramVerifyingConverter implements DatagramEndpoint {
+
+	private DatagramEndpoint endpoint;
+	private CertPathParameters pkiParms;
+	private CertPathValidator validator;
+
+	@Override
+	public void close() throws IOException {
+		endpoint.close();
+	}
+
+	@Override
+	public void send(DatagramMessage packet) throws IOException {
+		endpoint.send(packet);
+	}
+
+	private DatagramMessage internalReceive(DatagramMessage packet) throws IOException {
```

Comment:
internalReceive ist etwas wenig sprechend, wird hier nicht ein ASN1 Payload in eine DatagramMessage konvertiert oder so was?